lib/f_core/os/c_datalogger.cpp

```diff
@@ -0,0 +1,47 @@
+#include <f_core/os/c_datalogger.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(datalogger);
+
+namespace detail {
```

Comment:
Ooooooh. Is this so we save space?

Reply:
indeed. from some testing the template always gets inlined since its just a method call to this so we don't pay the code size price of 30 different CDataLoggers.